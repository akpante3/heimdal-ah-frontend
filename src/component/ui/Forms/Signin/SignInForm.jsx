import React from 'react';
import { Link } from 'react-router-dom';
import { FormGroup, Label, Input } from 'reactstrap';
import FormInput from '../../InputElements/FormInput';
import LoadingSpinner from '../../loadingSpinners/LoadingSpinner';
import SignInError from '../../errors/SignInError';
import './Signin.scss';

/**
 * @description - Heimdal Authors Haven Signin Component
 * @param {props} auth - the authentication state of the current user
 * @param {props} email - email value from
 * @param {props} errors - errors validations
 * @param {props} error - error validation from backend
 * @returns {component} form
 */
const SigninForm = (props) => {
  const {
    email,
    errors,
    error,
    password,
    onChange,
    isLoading,
    onLoginSubmit,
  } = props;

  return (
    <div className="heimdal-signin-form">
      <h1 className="form-title font-cardo text-center">Welcome back</h1>
      <form className="heimdal-form" onSubmit={onLoginSubmit} noValidate>
        <div className="font-cardo ph-30">
          <FormInput
            name="email"
            value={email}
            type="email"
            classname="form-control"
            placeholder="Email Address"
            onChange={onChange}
            errors={errors}
          />
          <FormInput
            name="password"
            value={password}
            type="password"
            classname="form-control"
            placeholder="Password"
            onChange={onChange}
            errors={errors}
          />
          <div className="d-flex justify-content-between mb-3">
            <div className="p-2">
              {' '}
              <FormGroup check>
                <Label check>
                  <Input type="checkbox" />
                  {' '}
                  Remember me
                </Label>
              </FormGroup>

            </div>
            <div className="p-2">
              {' '}
              <Link to='/'>Forgot password?</Link>
            </div>
          </div>
          <SignInError error={error} />
          <div className="row">
            <LoadingSpinner isLoading={isLoading} />
            <div className="col-md-8 text-left mb-10">
              <span className="text-muted p-t-10">
              Dont have an account?
                {' '}
                <Link to="/" className="link pt-2">
                Signup here
                </Link>
              </span>
            </div>
            <div className="col-md-4 text-right">
              <button type="submit" className="btn btn-default ph-25">
              Sign In
              </button>
            </div>
          </div>
        </div>
      </form>
      <br />
      <div className="text-center">
        <a
          href={`${process.env.BASE_URL_PROD}/auth/facebook`}
          className="btn btn-block btn-facebook"
        >
        Login with Facebook
        </a>
        <br />
        <a
          href={`${process.env.BASE_URL_PROD}/auth_twitter/twitter`}
          className="btn btn-block btn-twitter"
        >
        Login with Twitter
        </a>
        <br />
        <a href={`${process.env.BASE_URL_PROD}/auth/google`} className="btn btn-block btn-google">
        Login With Google
        </a>
      </div>
    </div>
  );
};

export default SigninForm;
