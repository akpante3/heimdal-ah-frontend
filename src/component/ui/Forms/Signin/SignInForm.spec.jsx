/* eslint-disable react/no-children-prop */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SignInForm from './SignInForm';

describe('The Signin Component Test Suite', () => {
  const wrapper = shallow(<SignInForm children={() => {}} />);
  it('should match the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // it('should handle change events', () => {
  //   const e = { target: { name: 'email', value: 'Heimdal' }, preventDefault: () => {} };
   
  //   wrapper.instance().props.onChange(e);
  //   expect(wrapper.instance().props.email).toEqual('Heimdal');
  //   // expect(wrapper.instance().email).toEqual(e.target.value);
  // });

 
});
