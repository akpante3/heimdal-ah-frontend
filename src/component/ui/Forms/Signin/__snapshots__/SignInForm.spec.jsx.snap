// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Signin Component Test Suite should match the snapshot 1`] = `
<div
  className="heimdal-signin-form"
>
  <h1
    className="form-title font-cardo text-center"
  >
    Welcome back
  </h1>
  <form
    className="heimdal-form"
    noValidate={true}
  >
    <div
      className="font-cardo ph-30"
    >
      <FormInput
        classname="form-control"
        name="email"
        placeholder="Email Address"
        type="email"
      />
      <FormInput
        classname="form-control"
        name="password"
        placeholder="Password"
        type="password"
      />
      <div
        className="d-flex justify-content-between mb-3"
      >
        <div
          className="p-2"
        >
           
          <FormGroup
            check={true}
            tag="div"
          >
            <Label
              check={true}
              tag="label"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <Input
                type="checkbox"
              />
               
              Remember me
            </Label>
          </FormGroup>
        </div>
        <div
          className="p-2"
        >
           
          <a
            href="/reset-password"
          >
            Forgot password?
          </a>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 text-center"
        >
          <LoadingSpinner
            isLoading={false}
          />
        </div>
        <MainError
          errors={Object {}}
        />
        <div
          className="col-md-8 text-left mb-10"
        />
        <div
          className="col-md-4 text-right"
        >
          <button
            className="btn btn-default ph-25 mx-r"
            type="submit"
          >
            Sign In
          </button>
        </div>
      </div>
    </div>
  </form>
  <br />
  <div
    className="text-center"
  >
    <a
      className="btn btn-block btn-facebook"
      href="undefined/api/v1/auth/facebook"
    >
      Login with Facebook
    </a>
    <br />
    <a
      className="btn btn-block btn-twitter"
      href="undefined/api/v1/auth_twitter/twitter"
    >
      Login with Twitter
    </a>
    <br />
    <a
      className="btn btn-block btn-google"
      href="undefined/api/v1/auth/google"
    >
      Login With Google
    </a>
  </div>
</div>
`;
